osu.Game/Rulesets/Difficulty/Skills/Skill.cs

```diff
@@ -61,19 +68,41 @@ public abstract class Skill
         protected Skill(Mod[] mods)
         {
             this.mods = mods;
+            Previous = new ReverseQueue<DifficultyHitObject>(PreviousCollectionSoftCapacity + 1);
         }
 
         /// <summary>
         /// Process a <see cref="DifficultyHitObject"/> and update current strain values accordingly.
         /// </summary>
         public void Process(DifficultyHitObject current)
         {
+            RemoveExtraneousHistory(current);
```

Comment:
Doesn't this only keep 1 item in the queue at all times (due to `while (Count > 1) Dequeue()`)? Whereas currently it's 2.

Reply:
Yes, but as it stands, the second object was actually never used in the codebase. It was likely just an artifact of the original calc being based on PP+ which used this for angle calculations.